platform/posix/transport/src/openssl_posix.c

```diff
@@ -623,10 +625,15 @@ int32_t Openssl_Recv( NetworkContext_t * pNetworkContext,
         }
         else
         {
-            LogError( ( "SSL_read of OpenSSL failed to receive data: "
-                        "status=%s.", ERR_reason_error_string( sslError ) ) );
+            LogError( ( "Failed to receive data over network: SSL_read failed: "
+                        "ErrorStatus=%s.", ERR_reason_error_string( sslError ) ) );
         }
     }
+    else if( pNetworkContext->pSsl == NULL )
+    {
+        LogError( ( "Failed to receive data over network: "
+                    "Network context has invalid SSL object." ) );
```

Comment:
Would it be better to say NULL instead of invalid?

Reply:
Updated